game/src/Game/Autoload/CursorManager.gd

```diff
@@ -0,0 +1,173 @@
+			frames = CursorSingleton.get_frames(cursor_name,index)
+			return
+
+		#couldnt find it, so generate it based on the highest res available
+		var highest_res_index : int = 0
+		var highest_res_x : int = 0
+		for i : int in range(len(resolutions)):
+			if resolutions[i].x > highest_res_x:
+				highest_res_x = resolutions[i].x
+				highest_res_index = i
+		generate_new_res(highest_res_index,resolution)
+		
+		self.resolutions = CursorSingleton.get_resolutions(self.cursor_name)
+		self.frames = CursorSingleton.get_frames(self.cursor_name,len(self.resolutions)-1)
+		self.hotspots = CursorSingleton.get_hotspots(self.cursor_name,len(self.resolutions)-1)
+		
+		assert(len(self.frames) != 0)
+
+	func generate_new_res(base_res_index:int, resolution:Vector2) -> void:
+		# resolution wasn't in among the default, need to generate it ourselves
+		CursorSingleton.generate_resolution(cursor_name,base_res_index,resolution)
+
+	#only bother with this if the cursor is animated
+	func _process_cursor(delta:float, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+		timeToFrame -= delta
+		if(timeToFrame <= 0):
+			currentFrame = (currentFrame + 1) % len(sequence)
+			timeToFrame = timings[self.sequence[currentFrame]] 
+			set_hardware_cursor(currentFrame, shape)
+
+	func set_hardware_cursor(frame:int=0, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+		var texture : ImageTexture = frames[sequence[frame]]
+		var hotspot : Vector2 = hotspots[sequence[frame]]
+		Input.set_custom_mouse_cursor(texture,shape,hotspot)
+
+
+#TODO: This is set on game start, but we probably want this to be a video setting
+var preferred_res : Vector2 = Vector2(32,32)
+
+
+var activeCursor : CompatCursor
+var activeShape : Input.CursorShape
+
+#Shape > Cursor dictionnaries
+var currentCursors : Dictionary = {
+	Input.CURSOR_ARROW:null,
+	Input.CURSOR_BUSY:null,
+	Input.CURSOR_IBEAM:null
+}
+var queuedCursors : Dictionary = {
+	Input.CURSOR_ARROW:null,
+	Input.CURSOR_BUSY:null,
+	Input.CURSOR_IBEAM:null
+}
+var loaded_cursors : Dictionary = {}
+
+func load_cursors() -> void:
+	CursorSingleton.load_cursors()
+	for cursor_name : StringName in CursorSingleton.cursor_names:
+		var cursor : CompatCursor = CompatCursor.new(cursor_name)
+		cursor.set_resolution(preferred_res)
+		loaded_cursors[cursor_name] = cursor
+
+var mouseOverWindow : bool = false
+
+#Handle queued cursor changes and cursor animations
+func _process(delta) -> void:
+	#only attempt to update the mouse when this wont crash anything
+	if mouseOverWindow:
+		var mouseShape : Input.CursorShape = Input.get_current_cursor_shape()
+
+		for shape in currentCursors.keys():
+			if currentCursors[shape] != queuedCursors[shape] and queuedCursors[shape] != null:
+				currentCursors[shape] = queuedCursors[shape]
+				currentCursors[shape].set_hardware_cursor(0,shape)
+			elif currentCursors[shape] != queuedCursors[shape]:
+				currentCursors[shape] = queuedCursors[shape]
+				Input.set_custom_mouse_cursor(null,shape)
```

Comment:
```suggestion
			if currentCursors[shape] != queuedCursors[shape]:
				currentCursors[shape] = queuedCursors[shape]
				if currentCursors[shape] != null:
					currentCursors[shape].set_hardware_cursor(0, shape)
				else:
					Input.set_custom_mouse_cursor(null, shape)
```
We can avoid repeating `currentCursors[shape] != queuedCursors[shape]` and also only write `currentCursors[shape] = queuedCursors[shape]` once